### PR TITLE
✨ Enable navigating pages through fragment params

### PR DESCRIPTION
**Pull request by Enriqe**
This PR enables support for navigating to arbitrary pages inside a story using a fragment params.

### CODE REVIEW

**Comment by Enriqe** [comment]
Need to test this a little bit more, will let you know when it's ready for review.

File: extensions/amp-story/1.0/test/test-amp-story.js [hunk 1]
```diff
@@ -1575,197 +1575,156 @@ describes.realWin(
           );
         });
       });
+    });
 
-      describe('amp-story branching', () => {
```
Comment (Enriqe) [inline extensions/amp-story/1.0/test/test-amp-story.js]: This `describe('amp-story branching', () => {  ...  }` was inside `describe('amp-story pause/resume callbacks', () => { ... }` which I don't think we want, so I just moved it out.

I also moved the toggling of the experiment from the before/afterEach and into the features that are still behind the experiment. The tests for the feature that this PR enables don't need the experiment anymore.

File: examples/amp-story/branching.html [hunk 2]
```diff
@@ -0,0 +1,78 @@
+<!doctype html>
```
Comment (gmajoulet) [inline examples/amp-story/branching.html]: `amp-story-branching.html`?

File: examples/amp-story/branching.html [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        poster-landscape-src="/examples/visual-tests/picsum.photos/image981_1600x900.jpg"
+        poster-square-src="/examples/visual-tests/picsum.photos/image981_1600x1600.jpg">
+      <amp-sidebar id="sidebar1" layout="nodisplay" >
+        <button id="close-button" on="tap:sidebar1.close">Close</button>
+        <ul>
+          <li><span><a href="/examples/amp-story/branching.html#page=page-2&foo=bar">Absolute fragment link</a></span></li>
+          <li><span><a href="#page=page-2">relative fragment link</a></span></li>
+          <li><a href="#page=page-3&foo=bar">In Link Hash</a></li>
+          <li><a href="/examples/amp-story/rtl.html#page=page-2">Link to another story using fragment</a></li>
+          <li on="tap:story.goToPage(id=page-3)">In Link Action</li>
```
Comment (gmajoulet) [inline examples/amp-story/branching.html]: Does this link work without the experiment enabled?

File: examples/amp-story/branching.html (hunk 2 above)
Comment (Enriqe) [reply examples/amp-story/branching.html]: We don't really follow this practice in this folder. Should we start using it?

File: examples/amp-story/branching.html (hunk 3 above)
Comment (Enriqe) [reply examples/amp-story/branching.html]: Nope, I forgot to remove it. Removed it now.

File: examples/amp-story/branching.html (hunk 2 above)
Comment (gmajoulet) [reply examples/amp-story/branching.html]: Ah right, that's fine :)

**Pull request merged**